### ISSUE TITLE
Can't find symbols for Microsoft.ML.Core or Microsoft.ML.Data

### ISSUE THREAD

**Issue by eerhardt** [dotnet/machinelearning#5552]
I tried debugging an application today that uses the latest release of ML.NET – 1.5.3. I believe this is the first release since the upgrade to Arcade.

However, I can’t load symbols for some of the assemblies. For the assemblies that are 1:1 with NuGet packages, those symbols appear to be loaded. But the core assemblies – ML.Core and ML.Data – don’t have their symbols loaded:

![image](https://user-images.githubusercontent.com/8291187/102251329-45c84200-3eca-11eb-9855-6e66b958543a.png)

We need to:

1. Upload the symbols for this release since it is an officially released version.
2. Fix the build so it automatically publishes the symbols for all assemblies.

cc @harishsk @michaelgsharp @mstfbl @frank-dong-ms

**Comment by antoniovs1029** [comment]
For the symbols you're actually seeing, is it possible to know when where they published?

**Comment by eerhardt** [comment]
> For the symbols you're actually seeing, is it possible to know when were they published?

I'm not sure on **when**. I assume that happened as part of the official build - it did before with the old build system.

As for **where** they are published, below is where VS finds them:

![image](https://user-images.githubusercontent.com/8291187/102264743-a9a73680-3edb-11eb-94b7-2c18514d17de.png)


```
C:\Program Files\dotnet\Microsoft.ML.CpuMath.pdb: Cannot find or open the PDB file.
C:\Users\eerhardt\.nuget\packages\microsoft.ml.cpumath\1.5.3\lib\netstandard2.0\Microsoft.ML.CpuMath.pdb: Cannot find or open the PDB file.
C:\WINDOWS\Microsoft.ML.CpuMath.pdb: Cannot find or open the PDB file.
C:\WINDOWS\symbols\dll\Microsoft.ML.CpuMath.pdb: Cannot find or open the PDB file.
C:\WINDOWS\dll\Microsoft.ML.CpuMath.pdb: Cannot find or open the PDB file.
C:\Users\eerhardt\AppData\Local\Temp\SymbolCache\Microsoft.ML.CpuMath.pdb\bafcdfc1f349489aadba25fff1af001affffffff\Microsoft.ML.CpuMath.pdb: Cannot find or open the PDB file.
C:\Users\eerhardt\AppData\Local\Temp\SymbolCache\MicrosoftPublicSymbols\Microsoft.ML.CpuMath.pdb\bafcdfc1f349489aadba25fff1af001affffffff\Microsoft.ML.CpuMath.pdb: Cannot find or open the PDB file.
C:\Users\eerhardt\AppData\Local\Temp\SymbolCache\Microsoft.ML.CpuMath.pdb\bafcdfc1f349489aadba25fff1af001a1\Microsoft.ML.CpuMath.pdb: Cannot find or open the PDB file.
C:\Users\eerhardt\AppData\Local\Temp\SymbolCache\MicrosoftPublicSymbols\Microsoft.ML.CpuMath.pdb\bafcdfc1f349489aadba25fff1af001a1\Microsoft.ML.CpuMath.pdb: Cannot find or open the PDB file.
SYMSRV:  BYINDEX: 0x3E
         C:\Users\eerhardt\AppData\Local\Temp\SymbolCache*https://msdl.microsoft.com/download/symbols
         Microsoft.ML.CpuMath.pdb
         BAFCDFC1F349489AADBA25FFF1AF001Affffffff


SYMSRV:  UNC: C:\Users\eerhardt\AppData\Local\Temp\SymbolCache\Microsoft.ML.CpuMath.pdb\BAFCDFC1F349489AADBA25FFF1AF001Affffffff\Microsoft.ML.CpuMath.pdb - path not found


SYMSRV:  UNC: C:\Users\eerhardt\AppData\Local\Temp\SymbolCache\Microsoft.ML.CpuMath.pdb\BAFCDFC1F349489AADBA25FFF1AF001Affffffff\Microsoft.ML.CpuMath.pd_ - path not found


SYMSRV:  UNC: C:\Users\eerhardt\AppData\Local\Temp\SymbolCache\Microsoft.ML.CpuMath.pdb\BAFCDFC1F349489AADBA25FFF1AF001Affffffff\file.ptr - path not found


SYMSRV:  HTTPGET: /download/symbols/Microsoft.ML.CpuMath.pdb/BAFCDFC1F349489AADBA25FFF1AF001Affffffff/Microsoft.ML.CpuMath.pdb


SYMSRV:  HttpQueryInfo: 801900c8 - HTTP_STATUS_OK


SYMSRV:  Microsoft.ML.CpuMath.pdb from https://msdl.microsoft.com/download/symbols: 18696 bytes 

SYMSRV:  PATH: C:\Users\eerhardt\AppData\Local\Temp\SymbolCache\Microsoft.ML.CpuMath.pdb\BAFCDFC1F349489AADBA25FFF1AF001Affffffff\Microsoft.ML.CpuMath.pdb


SYMSRV:  RESULT: 0x00000000


https://msdl.microsoft.com/download/symbols: Symbols downloaded from symbol server.
C:\Users\eerhardt\AppData\Local\Temp\SymbolCache\Microsoft.ML.CpuMath.pdb\BAFCDFC1F349489AADBA25FFF1AF001Affffffff\Microsoft.ML.CpuMath.pdb: Symbols loaded.
```

Note the final line: `https://msdl.microsoft.com/download/symbols: Symbols downloaded from symbol server.`. They are found on the official Microsoft Symbol Server.

**Comment by antoniovs1029** [comment]
The problem is that when arcade creates the Microsoft.ML.snupkg, it doesn't include any of the .PDBs of the projects it's meant to package. The Microsoft.ML.nupkg does include the correct DLLs from all its projects, but the symbols package doesn't include the .PDBs. This problem only happens on that package, and also for the CpuMathNative.pdb, which wasn't added to the Microsoft.ML.CpuMath.snupkg.

We still haven't found how to fix this, so we're still working on this.

**Comment by antoniovs1029** [comment]
We've fixed our symbols publishing pipeline in #5556 

Also, we've unlisted the 1.5.3 release from nuget.org, as there was another bug, and we've made a new release (1.5.4) that fixed that bug and include the symbols. We've checked that the 1.5.4 release does contain the symbols.

Thanks for noticing this.
